airflow/plugins_manager.py

```diff
@@ -142,41 +144,49 @@ def is_valid_plugin(plugin_obj, existing_plugins):
     for f in files:
         try:
             filepath = os.path.join(root, f)
-            if not os.path.isfile(filepath):
-                continue
-            mod_name, file_ext = os.path.splitext(
-                os.path.split(filepath)[-1])
-            if file_ext != '.py':
-                continue
-
-            log.debug('Importing plugin module %s', filepath)
-            # normalize root path as namespace
-            namespace = '_'.join([re.sub(norm_pattern, '__', root), mod_name])
-
-            m = imp.load_source(namespace, filepath)
-            for obj in list(m.__dict__.values()):
-                if is_valid_plugin(obj, plugins):
-                    plugins.append(obj)
-
-        except Exception as e:
+            try:
```

Comment:
Why do we have a nested try?

Reply:
That was one of the pylint errors - "filepath" which is printed in the first exception might not be set if there is an exception in os.path.join() :)